ngx_http_sxg_filter_module.c

```diff
@@ -706,11 +708,11 @@ static char* construct_fallback_url(const ngx_http_request_t* req) {
 
   if (req->args.len > 0) {
     snprintf(fallback_url, fallback_url_length, "%s%s%.*s?%.*s", kHttpsPrefix,
-             host, path_length, req->uri.data, (int)req->args.len,
+             host, path_length, req->unparsed_uri.data, (int)req->args.len,
              req->args.data);
   } else {
     snprintf(fallback_url, fallback_url_length, "%s%s%.*s", kHttpsPrefix, host,
```

Comment:
Is it possible to add support for unparsed_request_uri being an absolute URL? Or otherwise, just fall back to proxying the original content unsigned.

Per https://tools.ietf.org/html/rfc7230#section-5.3 we're supposed to support it, even though it'll rarely happen. So whatever's easiest, as long as it doesn't create something invalid like "https://https://site...".

Reply:
BTW, this is not a blocker for me. If it's not easy to do something for this case, we can file a bug.

Reply:
Sorry, I'm not sure what is specified as `unparsed_request_uri`.
Is that mean `req->uparsed_uri` contains `https://` started string? If so, can it be avoided with something like `if (memcmp(uri, "https://",..) == 0` ?
That should be fixed eventually but I think that should be done with another patch. Can you bug another issue?

Reply:
Yes, that's the idea. Filed #79.